openwisp-config/files/openwisp.init

```diff
@@ -1,119 +1,86 @@
+	add_option "$cfg" "--verify-ssl" verify_ssl
+	add_option "$cfg" "--uuid" uuid
+	add_option "$cfg" "--key" key
+	add_option "$cfg" "--shared-secret" shared_secret
+	add_option "$cfg" "--consistent-key" consistent_key
+	add_option "$cfg" "--hardware-id-script" hardware_id_script
+	add_option "$cfg" "--hardware-id-key" hardware_id_key
+	add_option "$cfg" "--bootup-delay" bootup_delay
+	add_option "$cfg" "--merge-config" merge_config
+	add_option "$cfg" "--test-config" test_config
+	add_option "$cfg" "--test-script" test_script
+	add_option "$cfg" "--connect-timeout" connect_timeout
+	add_option "$cfg" "--max-time" max_time
+	add_option "$cfg" "--capath" capath
+	add_option "$cfg" "--cacert" cacert
+	add_option "$cfg" "--mac-interface" mac_interface
+	add_option "$cfg" "--management-inteface" management_interface
+	add_option "$cfg" "--pre-reload-hook" pre_reload_hook
+	add_option "$cfg" "--post-reload-hook" post_reload_hook
+	add_option "$cfg" "--post-registration-hook" post_registration_hook
+}
 
 start_service() {
 	config_load openwisp
-	url=$(config_get http url)
-	interval=$(config_get http interval)
-	verify_ssl=$(config_get http verify_ssl)
-	uuid=$(config_get http uuid)
-	key=$(config_get http key)
-	shared_secret=$(config_get http shared_secret)
-	consistent_key=$(config_get http consistent_key)
-	hardware_id_script=$(config_get http hardware_id_script)
-	hardware_id_key=$(config_get http hardware_id_key)
-	bootup_delay=$(config_get http bootup_delay)
-	unmanaged=$(config_get http unmanaged)
-	merge_config=$(config_get http merge_config)
-	test_config=$(config_get http test_config)
-	test_script=$(config_get http test_script)
-	connect_timeout=$(config_get http connect_timeout)
-	max_time=$(config_get http max_time)
-	capath=$(config_get http capath)
-	cacert=$(config_get http cacert)
-	mac_interface=$(config_get http mac_interface)
-	management_interface=$(config_get http management_interface)
-	pre_reload_hook=$(config_get http pre_reload_hook)
-	post_reload_hook=$(config_get http post_reload_hook)
-	post_registration_hook=$(config_get http post_registration_hook)
-	if [ $url ]; then url="--url $url"; fi
-	if [ $interval ]; then interval="--interval $interval"; fi
-	if [ $verify_ssl ]; then verify_ssl="--verify-ssl $verify_ssl"; fi
-	if [ $uuid ]; then uuid="--uuid $uuid"; fi
-	if [ $key ]; then key="--key $key"; fi
-	if [ $shared_secret ]; then shared_secret="--shared-secret $shared_secret"; fi
-	if [ $consistent_key ]; then consistent_key="--consistent-key $consistent_key"; fi
-	if [ $hardware_id_script ]; then hardware_id_script="--hardware-id-script $hardware_id_script"; fi
-	if [ $hardware_id_key ]; then hardware_id_key="--hardware-id-key $hardware_id_key"; fi
-	if [ $bootup_delay ]; then bootup_delay="--bootup-delay $bootup_delay"; fi
+
+	# @todo This should be changed to a list element.
+	# So that we no longer need this workaround. And so we could also use
+	# the new add_option function. However, the agent must also be changed!
```

Comment:
I would not worry about this. This feature is probably going to be removed in the future.

Reply:
I would leave it as it is. If we refactor the service, we'll have to adjust this again.